faraday/proxy.py

```diff
@@ -19,21 +19,218 @@
 import sys
 import threading
 import time
+import argparse
+import shutil
 
 from flask import Flask
 from flask import request
 
 from faraday.uart import layer_4_service
 
 # Start logging after importing modules
-filename = os.path.abspath("loggingConfig.ini")
-logging.config.fileConfig(filename)
+
+relpath1 = os.path.join('etc', 'faraday')
+relpath2 = os.path.join('..', 'etc', 'faraday')
+setuppath = os.path.join(sys.prefix, 'etc', 'faraday')
+userpath = os.path.join(os.path.expanduser('~'), '.faraday')
+path = ''
+
+for location in os.curdir, relpath1, relpath2, setuppath, userpath:
+    try:
+        logging.config.fileConfig(os.path.join(location, "loggingConfig.ini"))
+        path = location
+        break
+    except ConfigParser.NoSectionError:
+        pass
+
 logger = logging.getLogger('Proxy')
+logger.debug("PATH: " + path)
+
+#Create Proxy configuration file path
+proxyConfigPath = os.path.join(path, "proxy.ini")
+logger.debug('Proxy.ini PATH: ', proxyConfigPath)
+
+# Command line input
+parser = argparse.ArgumentParser(description='Proxy application interfaces a Faraday radio over USB UART')
+parser.add_argument('--init-config', dest='init', action='store_true', help='Initialize Proxy configuration file')
+parser.add_argument('--callsign', help='Set Faraday callsign')
+parser.add_argument('--nodeid', type=int, help='Set Faraday node ID')
+parser.add_argument('--port', help='Set Faraday UART port')
+parser.add_argument('--baudrate', default='115200', help='Set Faraday UART baudrate')
+parser.add_argument('--timeout', type=int, default=5, help='Set Faraday UART timeout')
+parser.add_argument('--unit', type=int, default=0, help='Specify Faraday unit to configure')
+
+#Proxy options
+parser.add_argument('--number', type=int, default=0, help='Set number of Faraday radios to use')
```

Comment:
This is actually OK. Defaulting to 0 basically keeps the 1 in the `proxy.ini` file until you specify something other than 0 or 1 (since 0 is ignored... you'd never have 0 units to startup. Maybe in testmode but that's getting in the weeds for now). Programming a number.

This does allow negative numbers and any number that is positive